### PR TITLE
Add UI test for MenuItem about Space key

### LINKED ISSUE #11449: When we hit ENTER key to check/uncheck the checkbox present under 'ToolstripMenuItem1' dropdown is getting closed and keyboard focus loss is observed: A11y_.NET CoreWinforms_StripControls_Toolstripwithmenuitem_Keyboard
### .NET version

dotnet-sdk-9.0.100-preview.5.24258.1

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

_No response_

### GitHub Tags:
#A11yMAS; #A11yTCS; #A11ySev2; #BM_.NET Core[Winforms]_Win32_May2024; #.NETCore; #WCAG2.4.3; #DesktopApp; #Win11; #FTP; #Keyboard;

### Environment Details:
App name: .NET Core Winforms
Window Version: Win 11
OS Build: 22621.3085

### Repro Steps:

1. Launch the "Accessibility core app".
2. Main Form screen would appear.
3. TAB to "MenuToolbars_Controls:Testing the controls under Menu Toolbars Tab" and hit ENTER key.
4. TAB to StripControls button and hit ENTER key.
5. TAB to Toolstripwithmenuitem1 and hit ENTER key.
6. Try to check/ uncheck the checkbox using ENTER key and observe the issue.

### Actual Result:
When we hit ENTER key to check/uncheck the checkbox present under 'ToolstripMenuItem1' dropdown is getting closed and keyboard focus loss is observed.

### Expected Result:
When we hit ENTER key to check/uncheck the checkbox present under 'ToolstripMenuItem1' dropdown should not get closed and keyboard focus should remain on the checkbox.

### User Impact:
Keyboard users will face difficulty while interacting with the control as its getting closed and focus is moving to other controls.

### Attachment

https://github.com/dotnet/winforms/assets/90901519/2ed613c4-ecf8-4f9b-a331-e4b3c5c7d23c

### PR DESCRIPTION
**Pull request by SimonZhao888**
Fixes #11449

## Proposed changes

- Add UI tests for MenuItem about Space key.

## Test methodology 

- Automation test

 

## Test environment(s) 

- 9.0.100-preview.7.24407.12

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11964)

### CODE REVIEW

**Comment by Tanya-Solyanik** [comment]
@SimonZhao888  - is it possible to add some unit tests?

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStripDropDown.cs [hunk 1]
```diff
@@ -1251,8 +1251,12 @@ protected override void OnItemClicked(ToolStripItemClickedEventArgs e)
                 if ((!(e.ClickedItem is ToolStripDropDownItem dismissingItem))                   // it's not a dropdownitem
                    || (dismissingItem is ToolStripSplitButton && !dismissingItem.DropDown.Visible) // clicking on the split button button dismisses
                    || !(dismissingItem.HasDropDownItems))
-                {    // clicking on a item w/dropdown does not dismiss window
+                {   // clicking on a item w/dropdown does not dismiss window
                     Close(ToolStripDropDownCloseReason.ItemClicked);
+                    if (_ownerItem!.Owner!.Focused)
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStripDropDown.cs:1256]: Is it possible these will be null?

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStripMenuItem.ToolStripMenuItemAccessibleObject.cs [hunk 2]
```diff
@@ -146,6 +146,22 @@ internal override void Toggle()
                 _ => ToggleState.ToggleState_Indeterminate
             };
 
+        internal void OnCheckStateChanged(CheckState oldValue, CheckState newValue)
+        {
+            RaiseAutomationPropertyChangedEvent(
+                UIA_PROPERTY_ID.UIA_ToggleToggleStatePropertyId,
+                (VARIANT)(int)CheckStateToToggleState(oldValue),
+                (VARIANT)(int)CheckStateToToggleState(newValue));
+        }
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStripMenuItem.ToolStripMenuItemAccessibleObject.cs:155]:
```suggestion
        internal void OnCheckStateChanged(CheckState oldValue, CheckState newValue) =>
            RaiseAutomationPropertyChangedEvent(
                UIA_PROPERTY_ID.UIA_ToggleToggleStatePropertyId,
                (VARIANT)(int)CheckStateToToggleState(oldValue),
                (VARIANT)(int)CheckStateToToggleState(newValue));
```

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStripMenuItem.cs [hunk 3]
```diff
@@ -812,6 +814,12 @@ protected virtual void OnCheckedChanged(EventArgs e)
     protected virtual void OnCheckStateChanged(EventArgs e)
     {
         AccessibilityNotifyClients(AccessibleEvents.StateChange);
+        if (IsAccessibilityObjectCreated &&
+            AccessibilityObject is ToolStripMenuItemAccessibleObject accessibilityObject)
```
Comment (LeafShi1) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/ToolStrips/ToolStripMenuItem.cs:818]: if (IsAccessibilityObjectCreated 
     && AccessibilityObject is ToolStripMenuItemAccessibleObject accessibilityObject)

**Comment by Tanya-Solyanik** [comment]
@SimonZhao888  - @merriemcgaw will take this bug to the accessibility office hours to lower its priority.

**Comment by Tanya-Solyanik** [comment]
@SimonZhao888 @Olina-Zhang  - the bug had been resolved; do we still need this PR? However the test seems very useful, please adjust and keep it. We want to verify that space does not do anything.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MenuStripTests.cs [hunk 4]
```diff
@@ -820,6 +820,46 @@ public void MenuStrip_WndProc_InvokeMouseHoverWithHandle_Success()
...
+    {
+        using Form form = new Form();
+        using MenuStrip menuStrip = new() { TabStop = value };
+        using ToolStripMenuItem toolStripMenuItem1 = new();
+        using ToolStripMenuItem toolStripMenuItem2 = new() { CheckOnClick = true };
+        using ToolStripMenuItem toolStripMenuItem3 = new();
+        toolStripMenuItem1.DropDownItems.AddRange(toolStripMenuItem2, toolStripMenuItem3);
+        menuStrip.Items.AddRange(toolStripMenuItem1);
+        form.Controls.Add(menuStrip);
+        form.Show();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/MenuStripTests.cs:835]: Please move this test to S.W.F.UI.IntegrationTests assembly. All tests that show UI go there

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MenuStripTests.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace System.Windows.Forms.UITests;
+
+public class MenuStripTests
+{
+    [WinFormsTheory]
+    [BoolData]
+    public void MenuStrip_ProcessCmdKey_InvokeSpaceKey(bool value)
+    {
+        using Form form = new Form();
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MenuStripTests.cs:14]: `using Form form = new();`

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MenuStripTests.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        using MenuStrip menuStrip = new() { TabStop = value };
+        using ToolStripMenuItem toolStripMenuItem1 = new();
+        using ToolStripMenuItem toolStripMenuItem2 = new() { CheckOnClick = true };
+        using ToolStripMenuItem toolStripMenuItem3 = new();
+        toolStripMenuItem1.DropDownItems.AddRange(toolStripMenuItem2, toolStripMenuItem3);
+        menuStrip.Items.AddRange(toolStripMenuItem1);
+        form.Controls.Add(menuStrip);
+        form.Show();
+
+        Message m = new();
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MenuStripTests.cs:24]: ` Message message  = new();`

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MenuStripTests.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        if (value)
+        {
+            menuStrip.Focused.Should().BeTrue();
+        }
+        else
+        {
+            menuStrip.Focused.Should().BeFalse();
+        }
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MenuStripTests.cs:32]: `menuStrip.Focused.Should().Be(value);`

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MenuStripTests.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+        toolStripMenuItem3.ProcessCmdKey(ref m, keyData: Keys.Enter);
+        if (value)
+        {
+            menuStrip.Focused.Should().BeTrue();
+        }
+        else
+        {
+            menuStrip.Focused.Should().BeFalse();
+        }
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/MenuStripTests.cs:47]: `menuStrip.Focused.Should().Be(value);`

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**